src/main/java/org/wildfly/security/http/util/sso/DefaultSingleSignOnSession.java

```diff
@@ -160,10 +161,14 @@ public void put(SecurityIdentity identity) {
                             connection.getInputStream().close();
                         } catch (Exception cause) {
                             log.warnHttpMechSsoFailedLogoutParticipant(remoteURI.toString(), cause);
+                            // failed to logout participant, remove it from the list of participants
+                            try (SingleSignOn target = context.getSingleSignOnManager().find(id)) {
+                                target.removeParticipant(participantId);
+                            }
                         }
                     });
 
-                    try (SingleSignOn target = this.context.getSingleSignOnManagerManager().find(id)) {
+                    try (SingleSignOn target = this.context.getSingleSignOnManager().find(id)) {
                         if (target != null) {
                             // If all logout requests were successful, then there should be no participants, and we can invalidate the SSO
                             if (target.getParticipants().isEmpty()) {
```

Comment:
I think that what we can do is a better logging in cases that (eventually) the list of participants is not empty and we don't invalidate the SSO session. It will probably never happen, but better have some message there.